pennylane/qchem/tapering.py

```diff
@@ -290,28 +309,33 @@ def taper(h, generators, paulixops, paulix_sector):
 
     # cast to pauli sentence
     ps_u = pauli_sentence(u)
-    ps_h = pauli_sentence(h)
 
-    h = (ps_u * ps_h * ps_u).hamiltonian()  # cast back to hamiltonian
+    ps_h = h
+    if not isinstance(ps_h, qml.pauli.PauliSentence):
+        ps_h = pauli_sentence(h)
 
-    val = np.ones(len(h.terms()[0])) * complex(1.0)
+    ts_h = qml.pauli.PauliSentence()
+    for ps in _split_pauli_sentence(ps_h, max_size=PAULI_SENTENCE_MEMORY_SPLITTING_SIZE):
+        ts_h += ps_u * ps * ps_u  # helps restrict the peak memory usage for u @ h @ u
+    h = ts_h.hamiltonian()  # cast back to hamiltonian
 
     wireset = u.wires + h.wires
     wiremap = dict(zip(wireset, range(len(wireset) + 1)))
     paulix_wires = [x.wires[0] for x in paulixops]
 
-    for idx, w in enumerate(paulix_wires):
-        for i in range(len(h.terms()[0])):
-            s = qml.pauli.pauli_word_to_string(h.terms()[1][i], wire_map=wiremap)
-            if s[w] == "X":
-                val[i] *= paulix_sector[idx]
-
     o = []
+    h_coeffs, h_ops = h.terms()
+    val = np.ones(len(h_coeffs))
```

Comment:
What is this value?

Reply:
These take into account the sector value for the ground state if there's a PauliX on the Pauli term on the wires to be tapered. These are modified in the loop below here. 

```
[336]  for idx, w in enumerate(paulix_wires):
                if s[w] == "X":
                    val[i] *= paulix_sector[idx]
```

Reply:
So new coefficients?